### PR TITLE
Update account provider to allow direct inclusion of config and allow storage in .tfstate

### PR DESCRIPTION
**Pull request by lkwdwrd**
Still needs some additional testing, but initial tests are working well, so this is likely the initial state of the code with only minor updates coming.

### CODE REVIEW

File: docs/data-sources/tozny_realm_application_saml_description.md [hunk 1]
```diff
@@ -138,8 +138,9 @@ data "tozny_realm_application_saml_description" "saml_xml_description" {
 
 ### Top-Level Arguments
 
-* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the Terraform provider to use when fetching this application's SAML description.
-* `description` - (Computed) OIDC Client secret for the application.
+* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the Terraform provider to use when fetching this application's SAML description. Omit if using `client_credentials_config`.
+* `client_credentials_config` - (Optional) A JSON string containing Tozny client credentials for the provider to use
+* `description` - (Computed) OIDC Client secret for the application.s application's SAML description. Omit if using `client_credentials_filepath`.
```
Comment (galxy25) [inline docs/data-sources/tozny_realm_application_saml_description.md:143]: my bug, but description comment looks wrong as this is neither OIDC or a secret. Regarding the omit bit, you can use https://www.terraform.io/docs/extend/schemas/schema-methods.html `ConflictsWith` to enforce this "magic"

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 BINARY=terraform-provider-tozny
-VERSION=0.9.81
+VERSION=0.10.0
```
Comment (galxy25) [inline Makefile:2]: I usually do a find and replace across all *.tf files as well to auto update the examples to reference the latest version as well, but I can do that when its time to officially publish a new version

File: docs/resources/tozny_client_registration_token.md [hunk 3]
```diff
@@ -74,7 +74,8 @@ resource "tozny_client_registration_token" "realm_registration_token" {
...
 * `enabled` - (Optional) Whether the clients can be registered using this token. Defaults to true.
 * `one_time_use` - (Optional) Whether the token is only valid for registering a single client. Defaults to false.
-* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the provider to use when provisioning this realm. For this resource either this value or both `account_username` and `account_password` must be set on the provider.
+* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the provider to use when provisioning this realm. For this resource either this value or both `account_username` and `account_password` must be set on the provider. Omit if using `client_credentials_config`.
+* `client_credentials_config` - (Optional) A JSON string containing Tozny client credentials for the provider to use when provisioning this realm. For this resource either this value or both `account_username` and `account_password` must be set on the provider. Omit if using `client_credentials_filepath`.
```
Comment (galxy25) [inline docs/resources/tozny_client_registration_token.md:78]: `this realm` -> `this resource` (just so its generic, could also do `this registration token`

File: docs/resources/tozny_realm_application.md [hunk 4]
```diff
@@ -151,7 +151,8 @@ resource "tozny_realm_application" "aws_saml_application" {
 
 ### Top-Level Arguments
 
-* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the provider to use when provisioning this realm. For this resource either this value or both `account_username` and `account_password` must be set on the provider.
+* `client_credentials_filepath` - (Optional) The filepath to Tozny client credentials for the provider to use when provisioning this realm. For this resource either this value or both `account_username` and `account_password` must be set on the provider. Omit if using `client_credentials_config`.
```
Comment (galxy25) [inline docs/resources/tozny_realm_application.md:154]: this realm -> this resource

File: examples/accounts/.terraform.lock.hcl [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# This file is maintained automatically by "terraform init".
```
Comment (galxy25) [inline examples/accounts/.terraform.lock.hcl:1]: this should be git ignored

File: examples/realms/applications/clientSecret/.terraform.lock.hcl [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# This file is maintained automatically by "terraform init".
```
Comment (galxy25) [inline examples/realms/applications/clientSecret/.terraform.lock.hcl:1]: git ignore

File: tozny/data_source_tozny_realm_application_saml_description.go [hunk 7]
```diff
@@ -66,8 +77,7 @@ func dataSourceRealmApplicationSAMLDescription() *schema.Resource {
 func dataSourceRealmApplicationSAMLDescriptionRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 
-	toznyClientCredentialsFilePath := d.Get("client_credentials_filepath").(string)
-	toznySDK, err := MakeToznySDK(toznyClientCredentialsFilePath, m)
+	toznySDK, err := MakeToznySDK(d, m)
```
Comment (galxy25) [inline tozny/data_source_tozny_realm_application_saml_description.go:80]: more elegant then my approach, just pass in all the data and don't parse it in each resource

File: tozny/resource_account.go [hunk 8]
```diff
@@ -18,7 +17,7 @@ import (
 // account credentials suitable for
 // serializing to and from JSON for SDK consumption
 type AccountCredentialsFile struct {
-	APIEndpoit      string `json:"api_url"`
```
Comment (galxy25) [inline tozny/resource_account.go:21]: 🤦🏽

File: tozny/resource_account.go [hunk 9]
```diff
@@ -132,6 +131,13 @@ func resourceAccount() *schema.Resource {
 		ReadContext:   resourceAccountRead,
 		DeleteContext: resourceAccountDelete,
 		Schema: map[string]*schema.Schema{
+			"persist_credentials_to": {
+				Description: "Where to persist the generated account credentials. Default: none, they are not persisted.",
```
Comment (galxy25) [inline tozny/resource_account.go:135]: ValidateFunc function useful for enforcing valid enum value provided

https://www.terraform.io/docs/extend/schemas/schema-methods.html

File: tozny/resource_account.go [hunk 10]
```diff
@@ -154,6 +160,11 @@ func resourceAccount() *schema.Resource {
 				Default:     "tozny_client_credentials.json",
 				ForceNew:    true,
 			},
+			"config": {
+				Description: "the client configuration as a JSON string. Only populated with persist_credentails_to is set to 'terraform'",
```
Comment (galxy25) [inline tozny/resource_account.go:164]: `The`

**Review by galxy25 — APPROVED** [review]

**Comment by galxy25** [comment]
Excellent job refactoring and adding functionality in parallel!

File: tozny/resource_realm_broker_identity.go [hunk 11]
```diff
@@ -92,9 +101,9 @@ func resourceRealmBrokerIdentityCreate(ctx context.Context, d *schema.ResourceDa
 		return diag.FromErr(err)
 	}
 
-	realmBrokerIdedentityId := registeredBrokerIdentity.Identity.ToznyID.String()
+	realmBrokerIdedentityID := registeredBrokerIdentity.Identity.ToznyID.String()
```
Comment (efabens) [inline tozny/resource_realm_broker_identity.go:104]: `realmBrokerIdentitiyID` <- I think i spelled this right

**Review by efabens — APPROVED** [review]
One typo

**Comment by lkwdwrd** [comment]
Pushed an additional commit allowing the `persist_credentials_to` to get set on the broker resource as well. take a look at the last commit to see that code only. (specifically @galxy25 @efabens)

File: examples/accounts/main.tf [hunk 12]
```diff
@@ -27,14 +27,14 @@ resource "random_string" "account_username_salt" {
   special = false
 }
 
-# Generate a random string for use in creating
-# accounts across environments or executions conflict free
+# # Generate a random string for use in creating
```
Comment (galxy25) [inline examples/accounts/main.tf:30]: a few extra `# #`

**Review by efabens — APPROVED** [review]

**Review by galxy25 — APPROVED** [review]

**Pull request merged**